File: 1511-count-number-of-teams/count-number-of-teams.java
class Solution {
    public int numTeams(int[] rating) {
        int n = rating.length;
        int cnt = 0;

        for (int j = 1; j < n - 1; j++) {
            int lS = 0, lL = 0, rS = 0, rL = 0;

            for (int i = 0; i < j; i++) {
                if (rating[i] < rating[j]) {
                    lS++;
                } else {
                    lL++;
                }
            }

            for (int k = j + 1; k < n; k++) {
                if (rating[k] < rating[j]) {
                    rS++;
                } else {
                    rL++;
                }
            }

            cnt += lS * rL + lL * rS;
        }

        return cnt;
    }
}

File: 1511-count-number-of-teams/count-number-of-teams.py
class Solution(object):
    def numTeams(self, rating):
        n = len(rating)
        cnt = 0

        for j in range(1, n - 1):
            ls = ll = rs = rl = 0

            for i in range(j):
                if rating[i] < rating[j]:
                    ls += 1
                else:
                    ll += 1

            for k in range(j + 1, n):
                if rating[k] < rating[j]:
                    rs += 1
                else:
                    rl += 1

            cnt += ls * rl + ll * rs

        return cnt
